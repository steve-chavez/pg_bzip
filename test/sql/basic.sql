create extension if not exists bzip;

SELECT encode(bzcat(bzip2('this will be compresssed and then uncompressed')), 'escape') as result;

with repeated_str as (
  select repeat('some text', 100000) as val
)
select convert_from(bzcat(bzip2(val::bytea)), 'utf8') = val as succesful_compression_decompression
from repeated_str;
