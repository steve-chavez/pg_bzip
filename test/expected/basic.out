create extension if not exists bzip;
SELECT encode(bzcat(bzip2('this will be compresssed and then uncompressed')), 'escape') as result;
                     result                     
------------------------------------------------
 this will be compresssed and then uncompressed
(1 row)

with repeated_str as (
  select repeat('some text', 100000) as val
)
select convert_from(bzcat(bzip2(val::bytea)), 'utf8') = val as succesful_compression_decompression
from repeated_str;
 succesful_compression_decompression 
-------------------------------------
 t
(1 row)

